### DISCUSSION TITLE
Python API Doubts

### DISCUSSION THREAD

**Discussion by MayconSilva7** [JacquesLucke/animation_nodes#1912] [General]
Hi, 

I would like to know how to change the subprogram of "invoke subprogram". I have already a Script, I want to connect this script. 
I have been looking for invoke_subprogram.py in the Animation Nodes, and I have tried many functions, but I could not.

```
invoke_1 = test_node_tree.nodes.new("an_InvokeSubprogramNode")
invoke_1.subprogramNetwork()
```

I also tried the function create as follows:

```
invoke_1 = test_node_tree.nodes.new("an_InvokeSubprogramNode")
invoke_1.create("My Script")
```

I tried other functions, but nothing.

Thank you.

**Comment by MayconSilva7** [comment]
I got to resolve it, I saw your explanation here: https://github.com/JacquesLucke/animation_nodes/issues/1288

Thank you very much. Sorry to disturb you. I will delete the question.

**Comment by MayconSilva7** [comment]
Hi, @OmarEmaraDev. Look, Basically, I finished my code, I got to add and set all the nodes. Thank you. 

But Right now, I have a problem with **Invoke Subprogram**, because I need to link the **Generic List** output of one Invoke with another, as shown in the following image:

![image](https://user-images.githubusercontent.com/122628753/229098197-ef4eb128-43dd-4815-b6da-8268152069f7.png)

But, I received the following error: “Generic List not found”. 

Well, I tried to change the name, and other things but I did not get it. My objective is basically: to import from CSV file points to the Z coordinate of a plane. I mean, I want to generate my own displacement map as shown in the following image:

![image](https://user-images.githubusercontent.com/122628753/229100132-a22683bf-2bfc-4988-b2aa-2486c473a5b2.png)

I have been trying to import the **block text** to the Script node as shown in the following image, but I can not, I did not find it in the source code: 

![image](https://user-images.githubusercontent.com/122628753/229100685-b69f0009-2e41-4455-829c-702f2a3561fb.png)

As I have said, I got to add and set almost all the nodes, I am with a problem just with this one, and I do not know what I do. 

Sorry for one more question, and Thank you in advance.

**Reply by OmarEmaraDev** [reply]
Can you attach an example file with the code so that I can investigate it on my end?

**Comment by MayconSilva7** [comment]
Hi. First, thank you very much. I sent you an e-mail, in which I attached the Blender file with the code. There are two texts codes: 

Text: the script to use animation nodes

Text.001 - A random script to be imported 

Thank you very much in advance.

**Reply by OmarEmaraDev** [reply]
Sorry, I might have lost that email, can you attach the files here or resend it?

**Comment by MayconSilva7** [comment]


**Reply by OmarEmaraDev** [reply]
It seems you are trying to attach them using email, which will not work as far as I can see. So make sure to upload them in Github directly.

**Reply by MayconSilva7** [reply]
I send it to the e-mail shown in your profile.  GitHub doesn't support the Blend file. 

Thank you very much.

**Comment by OmarEmaraDev** [comment]
The invoke node inputs have special identifiers that do not match their displayed names, so you can just index them by their index.
That is:
```
test_node_tree.links.new(math.outputs["Result"], invoke_2.inputs[1])
```
